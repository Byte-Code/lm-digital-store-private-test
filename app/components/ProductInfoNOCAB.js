import React, { Component } from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import styled from 'styled-components';

const Wrapper = styled.div`
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  background-color: #efefef;
  padding: 69px 40px 57px;
  &>p {
    font-size: 18px;
    line-height: 28px;
    width: 425px;
    font-family: LeroyMerlinSans Light;
    width: 50%;
    &>span {
      font-family: LeroyMerlinSans;
    }
  }
`;

const Divider = styled.div`
  border: 1px dashed #67cb33;
  width: 100%;
  margin-top: 20px;
`;

export default class ProductInfoNOCAB extends Component {
  static propTypes = {
    descriptions: ImmutablePropTypes.list.isRequired
  }

  renderDescriptions() {
    const { descriptions } = this.props;
    if (!descriptions) {
      return null;
    }
    return descriptions.get(0).map(item => (
      <p key={`${item.get('label')}${item.get('value')}`}>
        <span>{item.get('label') && `${item.get('label')}: `}</span>
        {item.get('value')}
      </p>
    ));
  }

  render() {
    return (
      <Wrapper>
        {this.renderDescriptions()}
        <Divider />
      </Wrapper>
    );
  }
}
