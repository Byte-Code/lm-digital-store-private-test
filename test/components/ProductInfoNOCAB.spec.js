import React from 'react';
import { shallow } from 'enzyme';
import { List, fromJS } from 'immutable';

import ProductInfoNOCAB from '../../app/components/ProductInfoNOCAB';

const emptyDescriptions = List();
const descriptions = fromJS([
  [
    { label: 'label00', value: 'value00' },
    { label: 'label01', value: 'value01' },
    { label: 'label02', value: 'value02' }
  ],
  [
    { label: 'labelNR', value: 'valueNR' }
  ]
]);

describe('ProductInfoCAB', () => {
  it('should render no descriptions when they are not provided', () => {
    const result = shallow(
      <ProductInfoNOCAB
        descriptions={emptyDescriptions}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('should render just one list of descriptions', () => {
    const result = shallow(
      <ProductInfoNOCAB
        descriptions={descriptions}
      />
    );
    expect(result.find('p')).toHaveLength(3);
    expect(result).toMatchSnapshot();
  });
});
