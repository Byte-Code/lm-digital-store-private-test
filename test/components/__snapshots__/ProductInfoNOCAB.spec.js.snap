// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInfoCAB should render just one list of descriptions 1`] = `
<styled.div>
  <p>
    <span>
      label00: 
    </span>
    value00
  </p>
  <p>
    <span>
      label01: 
    </span>
    value01
  </p>
  <p>
    <span>
      label02: 
    </span>
    value02
  </p>
  <styled.div />
</styled.div>
`;

exports[`ProductInfoCAB should render no descriptions when they are not provided 1`] = `
<styled.div>
  <styled.div />
</styled.div>
`;
